raiden/utils/migrations/v18_to_v19.py

```diff
@@ -0,0 +1,218 @@
+
+
+def _add_blockhash_to_state_changes(storage: SQLiteStorage, cache: BlockHashCache) -> None:
+    """Adds blockhash to ContractReceiveXXX and ActionInitChain state changes"""
+
+    batch_size = 50
+    batch_query = storage.batch_query_state_changes(
+        batch_size=batch_size,
+        filters=[
+            ('_type', 'raiden.transfer.state_change.ContractReceive%'),
+            ('_type', 'raiden.transfer.state_change.ActionInitChain'),
+        ],
+        logical_and=False,
+    )
+    for state_changes_batch in batch_query:
+        # Gather query records to pass to gevent pool imap to have concurrent RPC calls
+        query_records = []
+        for state_change in state_changes_batch:
+            data = json.loads(state_change.data)
+            assert 'block_hash' not in data, 'v18 state changes cant contain blockhash'
+            record = BlockQueryAndUpdateRecord(
+                block_number=int(data['block_number']),
+                data=data,
+                state_change_identifier=state_change.state_change_identifier,
+                cache=cache,
+            )
+            query_records.append(record)
+
+        # Now perform the queries in parallel with gevent.Pool.imap and gather the
+        # updated tuple entries that will update the DB
+        updated_state_changes = []
+        pool_generator = Pool(batch_size).imap(
+            _query_blocknumber_and_update_statechange_data,
+            query_records,
+        )
+        for entry in pool_generator:
+            updated_state_changes.append(entry)
+
+        # Finally update the DB with a batched executemany()
+        storage.update_state_changes(updated_state_changes)
+
+
+def _add_blockhash_to_events(storage: SQLiteStorage, cache: BlockHashCache) -> None:
+    """Adds blockhash to all ContractSendXXX events"""
+    batch_query = storage.batch_query_event_records(
+        batch_size=500,
+        filters=[('_type', 'raiden.transfer.events.ContractSend%')],
+    )
+    for events_batch in batch_query:
+        updated_events = []
+        for event in events_batch:
+            data = json.loads(event.data)
+            assert 'triggered_by_block_hash' not in data, 'v18 events cant contain blockhash'
+            # Get the state_change that triggered the event and if it has
+            # a block number get its hash. If not fall back to latest.
+            matched_state_changes = storage.get_statechanges_by_identifier(
+                from_identifier=event.state_change_identifier,
+                to_identifier=event.state_change_identifier,
+            )
+            result_length = len(matched_state_changes)
+            msg = 'multiple state changes should not exist for the same identifier'
+            assert result_length == 1, msg
+
+            statechange_data = json.loads(matched_state_changes[0])
+            if 'block_hash' in statechange_data:
+                block_hash = statechange_data['block_hash']
```

Comment:
So, this is basically a rename?

Reply:
yeah the code looked different before. I can simplify.